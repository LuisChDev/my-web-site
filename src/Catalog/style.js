import styled from 'styled-components';

const StyledCatalog = styled.div`
  margin: 20px;
  padding: 20px;
  min-width: 50vw;
  overflow: hidden;
`;

const StyledButton = styled.button`
  border-color: #66023C;
  border-width: 5px;
  min-width: 100px;
  min-height: 50px;
`;

export {StyledCatalog, StyledButton};
