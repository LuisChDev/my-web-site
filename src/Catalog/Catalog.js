import React, {Component} from 'react';
import {StyledCatalog, StyledButton, StyledBts} from './style.js';
import Skill from './Skill/Skill.js';

/**
 * This component holds the technology skills I possess
 */
class Catalog extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeKey: 0,
      pose: "grown",
      turns: 0,
    };
    this.handleBack = this.handleClick.bind(this, false);
    this.handleForward = this.handleClick.bind(this, true);
  }

  render () {
    const {skills, buttons} = this.props;
    const {activeKey, pose, turns} = this.state;

    return (
      <StyledCatalog>
              <Skill
                name={skills[activeKey].name}
                description={skills[activeKey].description}
                logo={skills[activeKey].logo}
                pose={pose}
                turns={turns}
              />

        <StyledBts>
          <StyledButton onClick={this.handleBack}>
            {buttons.back}
          </StyledButton>
          <StyledButton onClick={this.handleForward}>
            {buttons.forward}
          </StyledButton>
        </StyledBts>
      </StyledCatalog>
    );
  }

  handleClick(forward) {
    let {skills} = this.props;
    let {activeKey, turns} = this.state;

    // proper modulus funcion
    function properMod(modulus, numb) {
      if (numb >= 0) {
        return numb%modulus;
      } else {
        while(numb < 0) {
          numb += modulus;
        } return numb;
      }
    }

    this.setState({
      pose: "shrunk",
    });

    setTimeout(() => {
      this.setState({
        turns: turns + (forward? 1:-1),
        activeKey: properMod(skills.length, activeKey + (forward? 1:-1)),
        pose: "grown",
      });
    }, 300);
  }
}

export default Catalog;
