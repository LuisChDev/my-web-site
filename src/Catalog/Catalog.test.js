import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Catalog from './Catalog.js';

describe('Catalog', () => {
  it("renders correctly", () => {
    const output = shallow(
        <Catalog skills={[
          {name: 'test1',
           description: 'lorem ipsum dolor sit amet',
           logo: 'logo1'},
          {name: 'test2',
           description: 'consectetur adipiscing elit',
           logo: 'logo2'}
        ]}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
