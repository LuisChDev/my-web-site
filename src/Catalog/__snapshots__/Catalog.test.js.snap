// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly 1`] = `
<styled.div>
  <Skill
    description="lorem ipsum dolor sit amet"
    logo="logo1"
    name="test1"
    pose="open"
  />
  <div>
    <styled.button
      onClick={[Function]}
    >
      Previous
    </styled.button>
    <styled.button
      onClick={[Function]}
    >
      Next
    </styled.button>
  </div>
</styled.div>
`;
